pkg/indexer/atlasdatabaseuserprojects.go

```diff
@@ -0,0 +1,48 @@
+package indexer
+
+import (
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+const (
+	AtlasDatabaseUserByProjectsIndex = "atlasdatabaseuser.spec.project"
```

Comment:
this name does not reflect what the index does. It does not only index using `atlasdatabaseuser.spec.project` but also the external atlas ID. I recommend splitting this into separate indexes.

Reply:
Agreed.